static/js/containers/CollectionDetailPage.js

```diff
@@ -21,6 +26,7 @@ class CollectionDetailPage extends React.Component {
     collectionKey: string,
     editable: boolean,
```

Comment:
I noticed that the gear icon is always enabled and will show the edit dialog even if the user does not have edit permission, but a 'You do not have permission' page will load after the form is submitted.  Ideally the gear icon would be disabled instead (@pdpinch maybe this should wait for a future PR?). 

The `CollectionReactView` would need to be modified to something like this on order to return the appropriate `editable` value if a detail page is being requested:
```python
    def get_context_data(self, **kwargs):  # pylint: disable=arguments-differ
        context = super().get_context_data(**kwargs)
        keymatch = re.search(r'/([0-9a-f]{32})/$', self.request.path)
        context["js_settings_json"] = json.dumps({
            **default_js_settings(self.request),
            'editable': keymatch and ui_permissions.has_admin_permission(
                Collection.objects.get(key=keymatch.group(1)), self.request)
        })
        return context
```
However I couldn't quite figure out how to pass that value through the router to the `CollectionDetailPage`

Reply:
If it can't be done today, we should leave this for another issue. 

Ideally, the gear icon shouldn't display at all for someone without admin permission.

Reply:
If `editable` is passed to the `CollectionDetailPage` props, the gear icon could be selectively displayed like this:

```jsx
    return <div className="centered-content">
      <header>
        ......
        {editable &&
        <div className="tools">
          <a onClick={R.partial(showDialog, [DIALOGS.NEW_COLLECTION])}>
            <i className="material-icons">settings</i>
          </a>
        </div>
        }
      </header>
      { this.renderCollectionVideos(videos) }
    </div>;
```

Reply:
Good catch, Matt. Returning the permission as part of the API call might also work. I'll tinker with it, but if it looks too heavy I'll create a separate issue